common/app/common/NewNavigation.scala

```diff
@@ -509,19 +509,19 @@ object NewNavigation {
 
     def getSectionOrTagId(page: Page) = {
       val tags = Navigation.getTagsFromPage(page)
-      val commonKeywords = tagPages.intersect(tags.keywordIds)
+      val commonKeywords = tags.keywordIds.intersect(tagPages).sortWith(tags.keywordIds.indexOf(_) < tags.keywordIds.indexOf(_))
```

Comment:
@DiegoVazquezNanini I completely agree, but I will do this tomorrow, and get this change out today - its a blocker on some coms to go out to editorial. 

A lot of this logic needs tests, but up until yesterday there was still the chance of throwing all of this away (we still needed approval once the test was successful). Now I really want to go through and write some tests and rewrite things, and document it all.